### PR TITLE
Table model insertion auto-cast & Fix TVList not removed when all data are deleted.

### PR DESCRIPTION
**Pull request by jt2594838**
Table model insertion can use auto-cast if the incoming data type is compatible with the registered type.
Below are the tolerant casts, the first column is the registered type, and the second column is the incoming type.
img_v3_02hf_cad6a09f-831d-4fa7-a7d0-176ebdc5aefg

### CODE REVIEW

File: iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties [hunk 1]
```diff
@@ -70,3 +70,121 @@ cn_metric_prometheus_reporter_port=9091
...
+#Thu Dec 12 10:40:33 HKT 2024
+#enable_partial_insert=true
+#Thu Dec 12 10:40:33 HKT 2024
+#enable_partial_insert=false
+#Thu Dec 12 10:40:37 HKT 2024
+#enable_partial_insert=true
+#Thu Dec 12 10:40:38 HKT 2024
+#enable_partial_insert=false
+#Thu Dec 12 10:42:19 HKT 2024
+#enable_partial_insert=true
```
Comment (HTHou) [inline iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties:84]: Should remove these?

Comment (jt2594838) [reply iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties]: Done

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/statement/crud/InsertRowStatement.java [hunk 2]
```diff
@@ -190,9 +190,8 @@ public Object getFirstValueOfIndex(int index) {
 
   @Override
   protected boolean checkAndCastDataType(int columnIndex, TSDataType dataType) {
-    if (CommonUtils.checkCanCastType(dataTypes[columnIndex], dataType)) {
-      values[columnIndex] =
-          CommonUtils.castValue(dataTypes[columnIndex], dataType, values[columnIndex]);
+    if (dataType.isCompatible(dataTypes[columnIndex])) {
+      values[columnIndex] = dataType.castFromSingleValue(dataType, values[columnIndex]);
```
Comment (luoluoyuyu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/statement/crud/InsertRowStatement.java:194]:
```suggestion
      values[columnIndex] = dataType.castFromSingleValue(dataTypes[columnIndex], values[columnIndex]);
```

Comment (jt2594838) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/statement/crud/InsertRowStatement.java]: Done

**Review by luoluoyuyu — APPROVED** [review]

**Review by Caideyipi — APPROVED** [review]

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**